### PR TITLE
Feature/apl 1497 bom

### PR DESCRIPTION
**Pull request by silaev-firstbridge**

### CODE REVIEW

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/http/API.java [hunk 1]
```diff
@@ -176,14 +179,34 @@ public API(PropertiesHolder propertiesHolder, UPnP upnp, JettyConnectorCreator j
...
-        String dir = DirProvider.getBinDir() + File.separator + WEB_UI_DIR;
-        dir = dir + File.separator + "build";
-        File res = new File(dir);
-        if (!res.exists()) { //we are in develop IDE or tests
-            dir = DirProvider.getBinDir() + "/apl-exec/target/" + WEB_UI_DIR + "/build";
-            res = new File(dir);
+        final Path binDir = DirProvider.getBinDir();
+
+        final Path htmlStubPath =
+            binDir.resolve("bin").resolve("html-stub").toAbsolutePath();
```
Comment (AndrewBoyarsky) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/http/API.java:185]: html stub is mandatory?

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/http/API.java [hunk 2]
```diff
@@ -176,14 +179,34 @@ public API(PropertiesHolder propertiesHolder, UPnP upnp, JettyConnectorCreator j
...
+        }
+
+        final String webUIProperty = propertiesHolder.getStringProperty("apl.webUIDir");
+        Path webUiPath;
+        try {
+            webUiPath = binDir.resolve(webUIProperty);
+
+            if (!Files.exists(webUiPath)
+                || !Files.isDirectory(webUiPath)
+                || !Files.exists(Path.of(webUiPath.toString(), "index.html"))
```
Comment (AndrewBoyarsky) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/http/API.java:198]: webUiPath.resolve("index.html") = Path.of(webUiPath.toString(), "index.html")

File: apl-core/src/main/resources/META-INF/beans.xml [hunk 3]
```diff
@@ -1,9 +1,13 @@
 <?xml version="1.0"?>
 <beans xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
        xmlns="http://xmlns.jcp.org/xml/ns/javaee"
+       xmlns:weld="http://jboss.org/schema/weld/beans"
        xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/javaee http://xmlns.jcp.org/xml/ns/javaee/beans_1_1.xsd"
        version="1.1" bean-discovery-mode="all">
   <decorators>
     <class>org.jboss.weld.environment.se.threading.RunnableDecorator</class>
   </decorators>
+  <weld:scan>
```
Comment (AndrewBoyarsky) [inline apl-core/src/main/resources/META-INF/beans.xml:10]: What is a reason to exclude it?

Comment (yuriy-larin-firstbridge) [reply apl-core/src/main/resources/META-INF/beans.xml:10]: What was attempt to fix error by me, when 'fat jar' is created and there was one error that can be fixed by that xml. Looks like that doesn't effect the node run.

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/http/API.java (hunk 1 above)
Comment (silaev) [reply apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/http/API.java:185]: yes, it's used if web-ui is not available

Comment (silaev-firstbridge) [reply apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/http/API.java:185]: yes, it's used if web-ui is not available

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/http/API.java (hunk 2 above)
Comment (silaev-firstbridge) [reply apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/http/API.java:198]: index.html is used in `Path.of(webUiPath.toString(), "index.html")`

File: apl-core/src/main/resources/META-INF/beans.xml (hunk 3 above)
Comment (yuriy-larin-firstbridge) [reply apl-core/src/main/resources/META-INF/beans.xml:10]: Now 'beans.xml' is updated a lot. There are many excludes in it for 'fat jar'.

**Pull request merged**